astropy/modeling/fitting.py

```diff
@@ -840,6 +838,61 @@ def _convert_input(x, y, z=None, n_models=1, model_set_axis=0):
     return farg
 
 
+class MinimizeFitter(Fitter):
+    """
+
+    A fitter for the minimize optimizer.
+
+    Parameters
+    ----------
+    method : str or callable(custom)
+        select minimization method
+
+    supported_constraints : list
+        lists the constraint types supported by the Optimization method default is fixed
+        and tied parameters, these can be set for custom methods for predifined methods
+        this will be changed atomatically
+    """
+
+    def __init__(self, method='slsqp', supported_constraints=None):
+        super(MinimizeFitter, self).__init__(optimizer=Minimize, statistic=leastsquare, method=method, supported_constraints=supported_constraints)
+
+    def __call__(self, model, x, y, z=None, weights=None, **kwargs):
+        """
+        Fit data to this model.
+
+        Parameters
+        ----------
+        model : `~astropy.modeling.FittableModel`
+            model to fit to x, y, z
+        x : array
+            input coordinates
+        y : array
+            input coordinates
+        z : array (optional)
+            input coordinates
+        weights : array (optional)
+            weights
+        kwargs : dict
+            optional keyword arguments to be passed to the optimizer or the statistic
+
+        Returns
+        -------
+        model_copy : `~astropy.modeling.FittableModel`
+            a copy of the input model with parameters set by the fitter
+        """
+        model_copy = _validate_model(model, self._opt_method.supported_constraints)
+
+        farg = _convert_input(x, y, z)
+        farg = (model_copy, weights, ) + farg
+
+        p0, _ = _model_to_fit_params(model_copy)
+
+        fitparams, self.fit_info = self._opt_method(self.objective_function, p0, farg, **kwargs)
+        _fitter_to_model_params(model_copy, fitparams)
+        return model_copy
```

Comment:
It would be nice not only to return `model_copy`, but also the `scipy.optimize.OptimizeResult` object, so that the user can get the uncertainties on the estimated parameters for free through the `hess_inv` attribute.